src/identities/identities.js

```diff
@@ -66,17 +64,21 @@ const Identities = async ({ keystore, path, storage, ipfs } = {}) => {
   /**
    * Creates an identity, adding it to storage.
    * @param {Object} options Various options for configuring a new identity.
-   * @param {string} [options.type=publickey] The type of provider to use for generating an identity.
+   * @param {Function} [options.provider=PublicKeyIdentityProvider()] An instance of the Provider to use for generating an identity, e.g. PublicKeyIdentityProvider({ keystore })
    * @return {module:Identities~Identity} An instance of identity.
    * @memberof module:Identities~Identities
    * @instance
    */
   const createIdentity = async (options = {}) => {
     options.keystore = keystore
+    const DefaultIdentityProviderType = getIdentityProvider('publickey')
+    const IdentityProvider = options.provider || DefaultIdentityProviderType({ keystore })
+    const identityProvider = await IdentityProvider()
+
+    if (!getIdentityProvider(identityProvider.type)) {
```

Comment:
We can do this check now with `IdentityProvider.type` before L76

Reply:
Wasn't able to implement this:

1. IdentityProvider.type <- this exists
2. ~~IdentityProviderPartial.type~~ <- Partial (I.e. `IdentityProvider(options)`  is passed in as the `provider` param and doesn't have a type
3. identityProviderInstance.type <- this exists

When testing whether the IdentityProvider is being used, 2 does not expose a type so using 3 is the only option to retrieve the type.

Reply:
Ah, ok. Then let’s leave it as is 👍